### PR TITLE
Add Swagger documentation

### PR DESCRIPTION
**Pull request by nvllsvm**
Adds Swagger/OpenAPI version 2.0 documentation.

**Changes**
Adds the SwaggerService from https://github.com/jellyfin/Emby/blob/7aa573d4e0fc231932ad3da4cdf15199cf97709a/Emby.Server.Implementations/Services/SwaggerService.cs

**Issues**
The spec is version 2.0 here so it will be a stop-gap for https://github.com/jellyfin/jellyfin/issues/614.

### CODE REVIEW

File: Emby.Server.Implementations/Services/SwaggerService.cs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+            return paths;
+        }
+
+        private Dictionary<string, SwaggerMethod> GetPathInfo(RestPath info)
+        {
+            var result = new Dictionary<string, SwaggerMethod>();
+
+            foreach (var verb in info.Verbs)
+            {
+                var responses = new Dictionary<string, SwaggerResponse>
```
Comment (Bond-009) [inline Emby.Server.Implementations/Services/SwaggerService.cs]:
```suggestion
                var responses = new Dictionary<string, SwaggerResponse>();
```
No need to take up 3 lines.

File: Emby.Server.Implementations/Services/SwaggerService.cs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+            var spec = new SwaggerSpec
+            {
+                schemes = new[] { "http" },
+                tags = GetTags(),
+                swagger = "2.0",
+                info = new SwaggerInfo
+                {
+                    title = "Jellyfin Server API",
+                    version = "1.0.0",
```
Comment (Bond-009) [inline Emby.Server.Implementations/Services/SwaggerService.cs]: We have a proper API version, right?

File: Emby.Server.Implementations/Services/SwaggerService.cs [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        public object Get(GetSwaggerSpec request)
+        {
+            return _spec ?? (_spec = GetSpec());
+        }
+
+        private SwaggerSpec GetSpec()
+        {
+            string host = null;
+            Uri uri;
+            if (Uri.TryCreate(Request.RawUrl, UriKind.Absolute, out uri))
```
Comment (Bond-009) [inline Emby.Server.Implementations/Services/SwaggerService.cs]:
```suggestion
            if (Uri.TryCreate(Request.RawUrl, UriKind.Absolute, out Uri uri))
```
Oneliners :)

File: Emby.Server.Implementations/Services/SwaggerService.cs (hunk 1 above)
Comment (nvllsvm) [reply Emby.Server.Implementations/Services/SwaggerService.cs]: @Bond-009 I'll be rewriting this to support OpenAPI spec 3.0. Style changes will be addressed then.

File: Emby.Server.Implementations/Services/SwaggerService.cs (hunk 3 above)
Comment (nvllsvm) [reply Emby.Server.Implementations/Services/SwaggerService.cs]: @Bond-009 I'll be rewriting this to support OpenAPI spec 3.0. Style changes will be addressed then.

**Comment by nvllsvm** [comment]
@taconaut [ReDoc](https://github.com/Rebilly/ReDoc) will be added at some point.

**Pull request merged**